app/services/control_rate_service.rb

```diff
@@ -82,6 +85,26 @@ def controlled_patients(time)
 
   private
 
+  def cache_key
+    "#{self.class}/#{region.model_name}/#{region.id}/#{date_or_range_cache_key}"
+  end
+
+  def date_or_range_cache_key
+    if range
+      "#{range.min.to_s(:iso8601)}/#{range.max.to_s(:iso8601)}"
+    else
+      date.to_s(:iso8601)
+    end
+  end
+
+  def cache_version
+    "#{region.updated_at.utc.to_s(:usec)}/#{CACHE_VERSION}"
```

Comment:
What's the reason for a microsecond-level precision here?

Reply:
I was just using a builtin format that is all numeric and easy to read if we ever need to inspect or debug these keys.